### PR TITLE
Fix for #218

### PR DESCRIPTION
**Pull request by cdancy**
6 failures running integration tests locally but they do not appear to be related to this change. All look to be coming from commands being run inside containers and returning non-zero exit codes.

### CODE REVIEW

**Comment by cdancy** [comment]
Well it passed here so I guess that's good ;)

File: src/main/java/com/github/dockerjava/api/model/LogConfig.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class LogConfig {
+    
+    @JsonProperty("Type")
+    public String type = "json-file";
+    
+    @JsonProperty("Config")
+    public List<String> config = null;
```
Comment (cdancy) [inline src/main/java/com/github/dockerjava/api/model/LogConfig.java]: It should be noted this variable does nothing through docker at the moment (and I vaguely remember talk of it being removed) and serves only to help tests run successfully.

File: src/main/java/com/github/dockerjava/api/model/LogConfig.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class LogConfig {
+    
+    @JsonProperty("Type")
+    public String type = "json-file";
```
Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/api/model/LogConfig.java]: There shouldn't be a default value.

File: src/main/java/com/github/dockerjava/api/model/LogConfig.java (hunk 1 above)
Comment (marcuslinke) [inline src/main/java/com/github/dockerjava/api/model/LogConfig.java]: According to the docs `config` must be a `Map<String, String>`

**Comment by marcuslinke** [comment]
It would be the best to have at least a test case for LogConfig in `CreateContainerCmdImplTest`. Any chance?

File: src/main/java/com/github/dockerjava/api/model/LogConfig.java (hunk 2 above)
Comment (cdancy) [inline src/main/java/com/github/dockerjava/api/model/LogConfig.java]: Gotcha. Removed.

File: src/main/java/com/github/dockerjava/api/model/LogConfig.java (hunk 1 above)
Comment (cdancy) [inline src/main/java/com/github/dockerjava/api/model/LogConfig.java]: Fixed.

**Comment by cdancy** [comment]
That's the thing: no matter what one puts in for values the command to create a container succeeds. If I put in "fake-log-driver" for type docker will not complain and just default to log-driver none. If I put in configs that have nothing to do with the type docker will just ignore them. 

So with that in mind I'm not sure what we could even test. I've made the requested changes and fleshed out the support a bit more for this option.

**Comment by marcuslinke** [comment]
I think a basic test could create a container with a valid log config (like `json-file`) and inspect the container afterwards to check if the settings are accepted by the docker host.

**Comment by marcuslinke** [comment]
Don't worry about failed intergration tests. CircleCI uses a quite old docker version where log config is not supported.

**Comment by cdancy** [comment]
I've added the test 'CreateContainerCmdImplTest.createContainerWithLogConfig' which sets LogConfig to 'none' and inspects the reponse to ensure that is what is actually set. Works here against my local docker instance. I added the ' @Test(groups = "ignoreInCircleCi")' annotation as I assume we don't want this run on CircleCI due to it using an older docker version.

**Comment by marcuslinke** [comment]
@cdancy I've modified your code so the logging type is an enum now. Thanks for your contribution!

**Comment by cdancy** [comment]
Good call. I was thinking of doing the same but wasn't sure if we should force the user to pass in the Enum on the LogConfig constructor/setters or do a check to ensure the type was supported.

**Pull request merged**